launcher/src/main/java/org/apache/spark/launcher/SparkLauncher.java

```diff
@@ -107,6 +123,27 @@ public static void setConfig(String name, String value) {
     launcherConfig.put(name, value);
   }
 
+  /**
+   * Specifies that Spark Application be stopped if current process goes away.
+   * It tries stop/kill Spark Application if {@link LauncherServer} goes away.
+   * @return This launcher.
+   */
+  public SparkLauncher stopIfInterrupted() {
```

Comment:
Similar to previous comments, I kinda prefer `stopOnShutdown` since "interrupted" is kind of ill-defined in this context.